### ISSUE TITLE
HTTP 413 Content Too Large - pomerium_identity_profile cookies have grown in size

### ISSUE THREAD

**Issue by kking124** [pomerium/pomerium#4627]
## What happened?

A change in either Pomerium or our upstream IDP (Cognito) has added a large amount of profile data to the pomerium cookies. This causes us to exceed the configured header size of the proxied application server which is currently set to 8K.

Specifically, `pomerium_identity_profile` cookies have appeared. These cookies did not exist or were trivial in size before (see following image for information on cookie sizes):

![image](https://github.com/pomerium/pomerium/assets/21278391/b3ea101d-617d-426e-bb9f-f90e4ee03043)
 
## What did you expect to happen?

We expected the number and size of Cookies to remain approximately the same.

## How'd it happen?

1. Unknown - no environment changes were made prior to the issue appearing.

## What's your environment like?

- Pomerium version (retrieve with `pomerium --version`): 0.22.2
- Server Operating System/Architecture/Cloud: AWS Fargate container running the `pomerium/pomerium:v0.22.2` image

we maintain a separate, dedicated pomerium authentication service, which redirects to our Cognito IDP.

## What's your config.yaml?

```config.yaml
AUTHENTICATE_SERVICE_URL: <REDACTED>
AWS_REGION: <REDACTED>
CERTIFICATE: <REDACTED>
CERTIFICATE_KEY: <REDACTED>
DATABROKER_STORAGE_CONNECTION_STRING: <REDACTED>
DATABROKER_STORAGE_TLS_SKIP_VERIFY: true
DATABROKER_STORAGE_TYPE: redis
IDP_CLIENT_ID: <REDACTED>
IDP_CLIENT_SECRET: <REDACTED>
IDP_PROVIDER: oidc
IDP_PROVIDER_URL: cognito_uri
IDP_SCOPES: openid,email
METRICS_ADDRESS: ":8080"
PROGRAMMATIC_REDIRECT_DOMAIN_WHITELIST: <REDACTED>
SHARED_SECRET: <REDACTED>
SIGNING_KEY: <REDACTED>
```

## What did you see in the logs?

```logs
N/A - no logs found related to this issue
```

## Additional context

The first user report occurred at 2023-10-11T18:37:00-0400. So far, I have not found any better information in the logs about when it began.

These cookies did not seem to be previously required for the functioning of pomerium, and were not needed by our application. If we could learn how/why this additional data is being sent we would like to stop sending it so we can reduce http request sizes.

It was suggested that the profile data could contain a profile image; this does not appear to be the case as users do not have the ability to set profile images in our Cognito infrastructure, currently.

**Comment by Phyxius** [comment]
FWIW I’m seeing issues with this too. It would be nice to have this configurable since many of the services I use don’t consume the cookie at all.

**Comment by Shiikos** [comment]
I also encountered the same issues. Occurs when upgrading Pomerium version 0.22.3 to Pomerium version 0.23.0.

**Comment by calebdoxsey** [comment]
Will be fixed by https://github.com/pomerium/pomerium/pull/4765

**Comment by desimone** [comment]
@kenjenkins  to add a small confirming unit/integration test though these cookies should not exist anymore as state data is no longer stored client side.

**Comment by desimone** [comment]
This should be fixed with #4765 
\cc @Phyxius \ @kking124
